client/state/posts/reducer.js

```diff
@@ -253,6 +254,9 @@ export function edits( state = {}, action ) {
 				}
 
 				return set( memoState, [ post.site_ID, post.ID ], omitBy( postEdits, ( value, key ) => {
+					if ( key === 'terms' ) {
```

Comment:
I'm not familiar with the API results, but this make me wonder if there's other keys that could have different shapes?